### PR TITLE
Added stub for sratools

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #XXX 

- [x] Added stub for sratools and avoided '.' in the path bug: https://github.com/ncbi/sra-tools/issues/865
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/sratools/fasterqdump/main.nf [hunk 1]
```diff
@@ -40,6 +43,28 @@ process SRATOOLS_FASTERQDUMP {
...
+        pigz: \$( pigz --version 2>&1 | sed 's/pigz //g' )
+    END_VERSIONS
+    """
+
+    stub:
+    def args2       = task.ext.args2    ?:  ''
+    def prefix      = task.ext.prefix   ?:  "${meta.id}"
+    def outfile     = meta.single_end   ?   "${prefix}.fastq" : "${prefix}_1.fastq ${prefix}_2.fastq"
+    """
+    touch $outfile
```
Comment (Midnighter) [inline modules/nf-core/sratools/fasterqdump/main.nf:66]: I find it convenient in a stub to echo the original command, so it shows what would have been executed.

File: modules/nf-core/sratools/fasterqdump/main.nf [hunk 2]
```diff
@@ -23,7 +23,10 @@ process SRATOOLS_FASTERQDUMP {
     def args = task.ext.args ?: ''
     def args2 = task.ext.args2 ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def outfile = meta.single_end ? "${prefix}.fastq" : prefix
+    def outfile = "${prefix}.fastq"
```
Comment (Midnighter) [inline modules/nf-core/sratools/fasterqdump/main.nf:26]: Why does this no longer distinguish between single-end and paired-end files? In the stub below you have
```groovy
def outfile = meta.single_end ? "${prefix}.fastq" : "${prefix}_1.fastq ${prefix}
```

I would think that `--outfile ${prefix}.fastq` for paired-end data causes confusion with filenames?

File: modules/nf-core/sratools/prefetch/main.nf [hunk 3]
```diff
@@ -32,4 +32,16 @@ process SRATOOLS_PREFETCH {
     }
 
     template 'retry_with_backoff.sh'
+
+    stub:
+    """
+    mkdir $id
```
Comment (Midnighter) [inline modules/nf-core/sratools/prefetch/main.nf:38]: Again, I like to echo the original command.

**Review by Midnighter — COMMENTED** [review]
Looks good overall 🙂. I have a few specific questions, but nothing major.

File: modules/nf-core/sratools/fasterqdump/main.nf (hunk 1 above)
Comment (GallVp) [reply modules/nf-core/sratools/fasterqdump/main.nf]: Thanks @Midnighter 

That's a good idea. I have updated to add echo.

File: modules/nf-core/sratools/fasterqdump/main.nf (hunk 2 above)
Comment (GallVp) [reply modules/nf-core/sratools/fasterqdump/main.nf:26]: fasterq-dump has two modes for specifying outfile prefix. The one which takes 'something' as prefix is currently broken due to '.' in the path bug. Please see: https://github.com/ncbi/sra-tools/issues/865

If we use 'something.fastq', luckily, the '.' in the path bug can be avoided. But this mode creates three files for some paired end SRAs with '.fastq' extension which are emitted by the process. To exclude the third file, `mv prefix.fastq prefix || echo '...'` is used.

File: modules/nf-core/sratools/prefetch/main.nf (hunk 3 above)
Comment (GallVp) [reply modules/nf-core/sratools/prefetch/main.nf:38]: `prefetch` is using a template for script. Is there a way to echo the template?

Comment (Midnighter) [reply modules/nf-core/sratools/prefetch/main.nf:38]: My bad, I had forgotten about that.

**Review by Midnighter — APPROVED** [review]
Thank you for answering my questions 👍🏼.

**Pull request merged**